src/Microsoft.AspNet.Mvc.Core/Formatters/InputFormatter.cs

```diff
@@ -62,12 +62,17 @@ public virtual bool CanRead(InputFormatterContext context)
 
             var contentType = context.HttpContext.Request.ContentType;
             MediaTypeHeaderValue requestContentType;
-            if (!MediaTypeHeaderValue.TryParse(contentType, out requestContentType))
+            if (!MediaTypeHeaderValue.TryParse(contentType, out requestContentType) || requestContentType == null)
```

Comment:
Reverted. I must have broken something elsewhere at an intermediate point in working on this issue. `TryParse()` can't return `true` with the `out` parameter left `null`.